### PR TITLE
[AOT] fix type in debug.ts

### PR DESCRIPTION
**Pull request by PPpro**
Re: https://github.com/cocos/3d-tasks/issues/16413

### Changelog

* 

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: cocos/core/platform/debug.ts [hunk 1]
```diff
@@ -49,80 +50,68 @@ let ccAssert = (condition: any, message?: any, ...optionalParams: any[]): void =
...
- * @param message @zh 包含零个或多个需要替换的JavaScript字符串。@en JavaScript objects to replace substitution strings in msg.
- * @param optionalParams  @zh 用来替换在message中需要替换的JavaScript对象。@en JavaScript objects with which to replace substitution strings within msg.
+ * @en Outputs a log message to the console. The message may be a single string (with optional substitution values), or it may be any one or more JavaScript objects.
+ * @zh 向控制台输出一条日志信息。这条信息可能是单个字符串（包括可选的替代字符串），也可能是一个或多个对象。
  */
-export function log (message?: any, ...optionalParams: any[]): void {
-    return ccLog(message, ...optionalParams);
+export function log (...data: unknown[]): void {
+    return ccLog(...data);
 }
```
Comment (PPpro) [inline cocos/core/platform/debug.ts:65]: Our implementation is aligned to web standards, so the `log` method should be defined as `function log (...data: unknown[]): void`, the same goes for other methods like `warn error debug`

File: cocos/core/utils/js-typed.ts [hunk 2]
```diff
@@ -348,7 +348,21 @@ const REGEXP_STR = /%s/;
...
+ * ```
+ * import { js } from 'cc';
+ * js.formatStr({}, null, undefined);  // [object Object] null undefined
+ * ```
+ */
+export function formatStr (...data: unknown[]): string
+export function formatStr (msg: unknown, ...subst: unknown[]): string {
     if (arguments.length === 0) {
         return '';
     }
```
Comment (PPpro) [inline cocos/core/utils/js-typed.ts]: we should support another formatStr signature, see this use case in unit test:
https://github.com/cocos/cocos-engine/blob/2a5b5711c1692c235bf05275b73cabacc9b15503/tests/core/js.test.ts#L41

**Comment by minggo** [comment]
Pleas send to v3.8.1.

**Comment by PPpro** [comment]
https://github.com/cocos/cocos-engine/pull/15426

**Pull request closed without merging**